src/go/rpk/pkg/cli/cmd/topic/consume.go

```diff
@@ -245,18 +256,28 @@ func consumeMessages(
 }
 
 func handleMessage(
-	msg *sarama.ConsumerMessage, mu *sync.Mutex, prettyPrint bool,
+	msg *sarama.ConsumerMessage, mu *sync.Mutex, prettyPrint, metaOnly bool,
 ) {
 	// Sometimes sarama will send nil messages.
 	if msg == nil {
 		log.Debug("Got a nil message")
 		return
 	}
+
+	var payloadPtr * string
+
+	var payload string
+	if !metaOnly {
+		payload = string(msg.Value);
+		payloadPtr = &payload
+	}
+
```

Comment:
Since `payload` is only used within the if block, there's no need to declare it outside.

```suggestion

	var payload *string

	if !metaOnly {
		p := string(msg.Value);
		payload = &p
	}
```

Reply:
Ha, had some difficulty changing mindset from C++ to Go.